### ISSUE TITLE
bug: `forge coverage --ir-minimum` compilation fails with  `Error: Unknown key "inliner` (solc < 0.8.5)

### ISSUE THREAD

**Issue by anajuliabit** [foundry-rs/foundry#9322]
### Component

Forge

### Have you ensured that all of these are up to date?

- [ ] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (25f24e6 2024-09-30T00:22:05.952969000Z) 

### What command(s) is the bug in?

forge coverage

### Operating System

macOS (Apple Silicon)

### Describe the bug

```
λ forge coverage --ir-minimum --skip test --skip script
Warning! "--ir-minimum" flag enables viaIR with minimum optimization, which can result in inaccurate source mappings.
Only use this flag as a workaround if you are experiencing "stack too deep" errors.
Note that "viaIR" is only available in Solidity 0.8.13 and above.
See more: https://github.com/foundry-rs/foundry/issues/3357
[⠊] Compiling...
[⠒] Compiling 311 files with Solc 0.8.19
[⠢] Compiling 43 files with Solc 0.6.12
[⠆] Compiling 12 files with Solc 0.8.10
[⠰] Solc 0.8.10 finished in 969.96ms
[⠔] Solc 0.6.12 finished in 614.18s
[⠒] Solc 0.8.19 finished in 614.78s
Error: 
Compiler run failed:
Error: Unknown key "inliner
```

I suspect that the foundry or some dependency is setting the `inliner` configuration for the compiler under the hood 

foundry.toml:

```toml
[profile.default]
src = 'src'
out = 'artifacts/foundry'
libs = ['node_modules', 'lib']
test = 'test'
cache_path  = 'artifacts/forge-cache'
fs_permissions = [{ access = "read", path = "./"}]
auto_detect_solc = true
evm_version = 'shanghai' # lock to shanghai to use push0 opcode which is live on base, optimism and Moonbeam
optimizer_runs = 1
rpc_storage_caching = { chains = "all", endpoints = "all" }
ignored_warnings_from = ["src/MErc20Delegator.sol", "src/tokensale/TokenSaleDistributorProxy.sol"]
gas_limit = "18446744073709551615" # u64::MAX
block_gas_limit = "18446744073709551615"
```

**Comment by grandizzy** [comment]
I am getting a seg fault on solc, so probably not too much we can do here, maybe you can reorg contracts to avoid such...
```
See more: https://github.com/foundry-rs/foundry/issues/3357
[⠰] Compiling...
[⠔] Compiling 428 files with Solc 0.8.19
[⠒] Compiling 43 files with Solc 0.6.12
[⠑] Compiling 12 files with Solc 0.8.10
[⠘] Solc 0.6.12 finished in 16.31ms
[⠒] Solc 0.8.10 finished in 1.03s
Error: solc exited with signal: 11 (SIGSEGV) (core dumped)
<empty output>
```

**Comment by zerosnacks** [comment]
Also running into the same 

```
Error: solc exited with signal: 11 (SIGSEGV) (core dumped)
<empty output>
```

@anajuliabit would you mind reporting this crash over at https://github.com/ethereum/solidity?

If possible please provide an alternative minimal reproduction in the meantime

For future reference: https://github.com/foundry-rs/compilers/blob/034ecd611eef030217c4b363a794226cd50b4b9e/crates/artifacts/solc/src/lib.rs#L713-L714

**Comment by grandizzy** [comment]
issue can be reproduce with any contract using solc <= 0.8.4, starting with 0.8.5 the following bug was addressed 
```Standard JSON: Properly allow the inliner setting under settings.optimizer.details.``` hence coverage works OK (see https://soliditylang.org/blog/2021/06/10/solidity-0.8.5-release-announcement/)
```Solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity 0.8.4;

contract AnotherCounter {
    
    function isContract(address account) internal view returns (bool) {
        bytes32 codehash;
        bytes32 accountHash = 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470;
        assembly {
            codehash := extcodehash(account)
        }
        return (codehash != accountHash && codehash != 0x0);
    }
}
```

In compilers we're setting inliner to false, hence the failure https://github.com/foundry-rs/compilers/blob/b5c2a49df8d271a9d0d3f851fc4f0023c1e29c9e/crates/artifacts/solc/src/lib.rs#L468-L471
Probably should be set to None.

**Comment by grandizzy** [comment]
not sure if we should take any action here though the message error is misleading. viaIR with minimum optimization is only available in Solidity 0.8.13 and above and we exit with err message if so, but only if version is auto detected

https://github.com/foundry-rs/foundry/blob/44c86e76d5e37505cd7349a867e64a845b5b9b2d/crates/forge/bin/cmd/coverage.rs#L97-L106

If we cannot detect version then coverage continues and for versions < 0.8.5 compilation fails with `Error: Unknown key "inliner`. Could sanitize settings to 0.8.4 if version cannot be detected but not sure a good idea, @zerosnacks @DaniPopes any thoughts on this? thanks

**Comment by zerosnacks** [comment]
Solady uses `^0.8.4` but from a user's perspective `^0.8.13` is where `via-ir` [was marked as production ready](https://soliditylang.org/blog/2022/03/16/solidity-0.8.13-release-announcement/) and became usable as it prevents the majority of stack-to-deep errors.

I think it would make sense to sanitize as suggested given the scope is limited to `forge coverage`.

I don't think there is anything inherently limiting this to `^0.8.13` given the`--ir-minimum` profile is something defined by us, not passed through to `solc`:

https://github.com/foundry-rs/compilers/blob/2db27d74ef5739da47ccf88ab3146b94462d3810/crates/artifacts/solc/src/lib.rs#L463-L491
